index.js

```diff
@@ -23,38 +25,45 @@ client.once('ready', () => {
 // this event will trigger each time a message is sent
 client.on('message', event => {
   
-
   // Check if the message start with our prefix and the author is not the bot
-  if (event.content.startsWith(Config.prefix) && event.author.id != client.user.id) {
-    console.log("-------------");
-    console.log("Command :\t", event.content);
-
-    // Slit the message on each whitespace
-    let splitedMessage = event.content.substring(Config.prefix.length).split(' ');
-
-    // Save the first element of the array as command
-    // Transform the string to lower case: EcHO -> echo
-    let command = splitedMessage[0].toLowerCase();
-
-    // Save all the element after the first one as args
-    let argsRaw = splitedMessage.splice(1);
-    let args = Utils.argsParser(argsRaw);
-
-    // If the command exist in the object Commands, execute it
-    if ( command in Commands ) {
-      Commands[command]({
-        event,
-        args,
-        argsRaw,
-        client
-      });
-    } else {
-      event.channel.send('Raté cette commande n\'existe pas camarade.');
-    }
+  if (!event.content.startsWith(Config.prefix)) return;
+  if (event.author.bot) return;
+
+  console.log("-------------");
+  console.log("Command :\t", event.content);
+
+  // Slit the message on each whitespace
+  let splitedMessage = event.content.substring(Config.prefix.length).split(' ');
 
-    console.log("\n\n");
+  // Save the first element of the array as commandName
+  // Transform the string to lower case: EcHO -> echo
+  let commandName = splitedMessage[0].toLowerCase();
 
+  // Save all the element after the first one as args
+  let argsRaw = splitedMessage.splice(1);
+  let args = Utils.argsParser(argsRaw);
+
+  // Find the command by its name or aliases
+  let commandObject = commands.find((i) => {
+    return i.name == commandName || i.aliases.includes(commandName)
+  });
+
+
+  if(typeof commandObject === 'undefined') {
+    event.channel.send('Raté cette commande n\'existe pas camarade.');
+  } else {
+    let opts = {
+      event,
+      args,
+      argsRaw,
+      client,
+      Discord,
+      commands,
+      Config
+    }
```

Comment:
@Fabio-web  *opts* argument now pass Discord.js, commands, and Config objects to your command so no need to require Discord.js inside your file anymore.

Reply:
Thank you 👍